### PR TITLE
ADT Event route e2e tests

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/resources/session-records/componentLifcycleTest.json [hunk 1]
```diff
@@ -1,172 +0,0 @@
-{
```
Comment (azabbasi) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/resources/session-records/componentLifcycleTest.json:1]: Is this intentional?

Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/resources/session-records/componentLifcycleTest.json:1]: Yeah, the title of this file is incorrect, and it doesn't correspond to a test that we have. We still have the "componentLifecycleTest.json" file

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        for (String eventRouteId : currentEventRouteIds) {
+            logger.info("Deleting event route " + eventRouteId + " before running the next test");
+            client.deleteEventRoute(eventRouteId);
+        }
+    }
+
+    // Note that only service returned eventRoute instances have their Id field set. When a user builds an instance locally,
+    // there is no way to assign an Id to it.
+    protected static void assertEventRoutesEqual(EventRoute expected, String expectedId, EventRoute actual) {
+        assertEquals(expectedId, actual.getId());
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java:56]: how do you feel about using assertj's fluent assertions here?

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java (hunk 2 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java:56]: I'm not sure that would simplify anything in this case. Am I missing something though?

Comment (abhipsaMisra) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java:56]: I feel it is more readable, and adds better context to the assertion being done.

Comment (abhipsaMisra) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java:56]: You could also do something like:
```java
assertThat(actual).extracting("id", "endpointName", "filter")
                 .doesNotContainNull()
                 .containsExactly(expectedId, expected.getEndpointName(), expected.getFilter());
```

Comment (abhipsaMisra) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/EventRoutesTestBase.java:56]: FYI - https://stackoverflow.com/questions/47969970/assertj-for-a-pojo-how-to-check-each-nested-property-field-in-one-chained-sente?answertab=votes#tab-top

**Pull request merged**